src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const ResultObject = [];
+
+  for (const operation of actions) {
+    const CurrentObject = {};
+
+    if (ResultObject.length !== 0) {
+      Object.assign(CurrentObject, ResultObject[ResultObject.length - 1]);
+    }
+
+    if (ResultObject.length === 0) {
+      Object.assign(CurrentObject, state);
+    }
+
+    if (operation.type === 'addProperties') {
+      Object.assign(CurrentObject, operation.extraData);
+      ResultObject.push(CurrentObject);
```

Comment:
don't do it in every `if`/`case` - just do it at the end of each loop iteration
push a copy of the object